taui/src/components/form.js

```diff
@@ -136,15 +143,25 @@ class Form extends React.PureComponent<Props> {
     const {destination, network} = this.state
     const destinations: Array<AccountAddress> = userProfile ? userProfile.destinations : []
     const locations = destinations.map(d => {
+      // check for older profiles that contain destination purposes not saved with
+      // the 'TripPurpose.' prefix required by react-i18next
+      const translatablePurpose = d.purpose.startsWith('TripPurpose')
+        ? d.purpose
+        : 'TripPurpose.' + d.purpose
       return {
-        label: d.purpose + ': ' + d.location.label,
+        label: d.location.label,
         position: d.location.position,
-        value: d.location.position
+        value: translatablePurpose
       }
     })
-    const destinationFilterOptions = createDestinationsFilter(locations)
+    const locationsWithLabels = locations.map(loc => {
+      // generate temporary, translated destination labels menu options
+      return {...loc, label: t(loc.value) + ': ' + loc.label}
+    })
+    const destinationFilterOptions = createDestinationsFilter(locationsWithLabels)
     const useNetworks = this.getProfileNetworks(this.props.networks, userProfile)
-    const networks = useNetworks.map(n => ({label: n.name, value: n.url}))
+    // generate temporary, translated network labels for menu options
+    const networks = useNetworks.map(n => ({label: t('Map.NetworkOptions.' + n.name.split(' ').join('')), value: n.url, networkName: n.name}))
```

Comment:
It would be worth checking to see if you can add a `networkKey` to the network configuration in `taui/configurations/default/store.yml` -- that defines the initial state of the Redux state, so if it works to add `networkKey` there, you should be able to define the translation key at that point, which would avoid the need to strip spaces from the network's `name` field in order to get the appropriate translation key.

In general, if you find yourself manipulating or parsing strings that were intended for human consumption in order to transform them into something programmatically meaningful, that's a pretty strong signal that something should be changed about the application's data model. The reason for this rule of thumb is that parsing human-readable strings often creates implicit requirements that can lead to errors or frustration down the line. In this case, stripping the spaces creates an implicit requirement that "all network translation keys must be exactly equal to the name of the network with internal spaces removed" which could be a source of confusion for someone trying to add a new network in the future if they don't know about that rule.

All that said, if it's not straightforward to add a new field to the network objects from the get-go for some reason, then this may be the best we can do.

Reply:
This has been surprisingly difficult to accomplish. I added a `networkKey` value to both `store.yml` and `taui/config.json`; however, somewhere in between state and the `Form` component, it loses the property. I can see that in `taui/src/actions/network.js`, the property exists on `state.data.networks`, but it does not exist on `json.networks` or on `c.networks` as a part of the `dispatch` action on line 90. My thought was perhaps it had something to do with what was being stored in local storage, which I tested by logging out and therefore triggering `clearLocalStorage`; that didn't work, so I thought perhaps it was something unique to configurations attached to the `TAUI_CONFIG_KEY` and other profile keys that were being passed into `retrieveConfig` in `network.js`. I tested this theory by creating a new profile with the new property on the networks, but that also did not work. Do you have any suggestions on what to try next?

Reply:
Thanks for digging in! I think this definitely qualifies at "not straightforward" at this point -- it looks like the network data from those files is getting swapped out with data from another source (probably the `.dat` files that it downloads, though I haven't confirmed that), so we'll have to figure out where this is happening and then how to thread through the data from the configuration. Let's make a technical debt issue for it, add a TODO to the code and revisit it toward the end of the project if we have time.

Reply:
Ok, great--I can make that technical debt issue now. Thank you!